File: frontend/src/types/fraud.ts
export type FraudDetectionType =
  | 'Duplicate Payment'
  | 'Amount Anomaly'
  | 'Vendor Anomaly'
  | 'Repeated Settlement Issue';

export type FraudRiskLevel = 'Critical' | 'High' | 'Medium' | 'Low';

export interface FraudAlertDTO {
  alert_id: string;
  txn_id: string;
  date: string;
  entity_name: string;
  amount: number;
  detection_type: FraudDetectionType;
  triggered_rule: string;
  risk_score: number;
  risk_level: FraudRiskLevel;
  reason: string;
  recommendation: string;
  status: 'Active' | 'Under Review' | 'Blocked' | 'Cleared';
  channel?: string;
  monetary_exposure: number;
  evidence_trail: string[];
}

export interface FraudTimelineEventDTO {
  id: string;
  timestamp: string;
  event_title: string;
  detection_type: FraudDetectionType;
  risk_level: FraudRiskLevel;
  description: string;
  monetary_impact: number;
  action_taken: string;
}

export interface FraudSummaryMetricsDTO {
  total_transactions_scanned: number;
  active_alerts_count: number;
  critical_alerts_count: number;
  high_alerts_count: number;
  medium_alerts_count: number;
  low_alerts_count: number;
  total_exposure_at_risk: number;
  prevented_loss_amount: number;
  anomaly_detection_rate_pct: number;
  by_detection_type: Record<string, number>;
}

export interface FraudCenterResponseDTO {
  summary: FraudSummaryMetricsDTO;
  alerts: FraudAlertDTO[];
  timeline: FraudTimelineEventDTO[];
}

File: backend/app/schemas/fraud.py
from typing import List, Dict, Optional
from pydantic import BaseModel, Field

class FraudAlertDTO(BaseModel):
    alert_id: str
    txn_id: str
    date: str
    entity_name: str
    amount: float
    detection_type: str  # "Duplicate Payment" | "Amount Anomaly" | "Vendor Anomaly" | "Repeated Settlement Issue"
    triggered_rule: str  # e.g., "Amount > 3x historical average" | "Duplicate settlement ID" | "Unusual frequency"
    risk_score: int      # 0 to 100
    risk_level: str      # "Critical" | "High" | "Medium" | "Low"
    reason: str
    recommendation: str
    status: str          # "Active" | "Under Review" | "Blocked" | "Cleared"
    channel: Optional[str] = "Bank Feed"
    monetary_exposure: float
    evidence_trail: List[str] = Field(default_factory=list)

class FraudTimelineEventDTO(BaseModel):
    id: str
    timestamp: str
    event_title: str
    detection_type: str
    risk_level: str
    description: str
    monetary_impact: float
    action_taken: str

class FraudSummaryMetricsDTO(BaseModel):
    total_transactions_scanned: int
    active_alerts_count: int
    critical_alerts_count: int
    high_alerts_count: int
    medium_alerts_count: int
    low_alerts_count: int
    total_exposure_at_risk: float
    prevented_loss_amount: float
    anomaly_detection_rate_pct: float
    by_detection_type: Dict[str, int]

class FraudCenterResponseDTO(BaseModel):
    summary: FraudSummaryMetricsDTO
    alerts: List[FraudAlertDTO]
    timeline: List[FraudTimelineEventDTO]

class UpdateFraudAlertStatusRequestDTO(BaseModel):
    alert_id: str
    new_status: str  # "Blocked" | "Cleared" | "Under Review"
    operator_notes: Optional[str] = None
